### PR TITLE
JIRA TS-3164 update redis version and throttling the request if redis is behind

### PR DESCRIPTION
**Pull request by georgeye**

### CODE REVIEW

File: timer_scheduler.js [hunk 1]
```diff
@@ -14,7 +15,12 @@ function handle_timeout() {
   var queues = get_queues(queue_daily, queue_by_hour);
   for(var i = 0; i < queues.length; i++) {
     utils.logDebug("going to fetch expired event from " + queues[i]);
-    expire_events(queues[i]);
+    if(work_client.command_queue_length > 10 || work_client.offline_queue_length > 10) {
```
Comment (nitzanharel) [inline timer_scheduler.js]: how is this slowing down?

Comment (georgeye) [inline timer_scheduler.js]: there are are on flights request, we skip this round of scanning of expired entries. thus giving more time for redis to handle this pending requests.

Comment (nitzanharel) [inline timer_scheduler.js]: then why is this checked per queue an not before you even look at the queues ?

Comment (georgeye) [inline timer_scheduler.js]: that is a good optimization we can put in.

**Pull request merged**